packages/core/src/agent/index.ts

```diff
@@ -1356,11 +1379,12 @@ export class Agent<
             title: thread.title,
             memoryConfig,
             resourceId,
+            saveThread: false,
           });
         }
 
         let [memoryMessages, memorySystemMessage] =
-          thread.id && memory
+          thread.id && memory && existingThread
```

Comment:
I know this wasn't your change but to make it more clear what thread.id vs threadId is
```suggestion
          threadId && memory && existingThread
```

Reply:
Actually we can probably just check if there's an existing thread, the other two will be true if there's a thread